repairnator/repairnator-dockerpool/src/main/java/fr/inria/spirals/repairnator/dockerpool/AbstractPoolManager.java

```diff
@@ -93,20 +93,32 @@ public void setEngines(List<SerializerEngine> engines) {
         this.engines = engines;
     }
 
-    public RunnablePipelineContainer submitBuild(String imageId, int buildId) {
+    public TreatedBuildTracking prepareBeforeSubmitBuild(int buildId) {
         this.cleanUpOlderContainers();
-        TreatedBuildTracking treatedBuildTracking = null;
         try {
-            treatedBuildTracking = new TreatedBuildTracking(this.engines, this.runId, buildId);
+            return new TreatedBuildTracking(this.engines, this.runId, buildId);
         } catch (IOException e) {
```

Comment:
Actually the IOException is useless now: it should be completely removed. It'll avoid we got a NPE after.

Reply:
But if I remove the try-catch block, IntelliJ tells me `Unhandled exception: java.io.IOException`.

Reply:
Yup. You have to remove the `throws IOException` from TreatedBuildTracking constructor: it's not used.

Reply:
Oh, ok. Done.